src/Sinch/Conversation/TemplatesV2/TemplatesV2.cs

```diff
@@ -0,0 +1,190 @@
+using System.Web;
+using Sinch.Core;
+using Sinch.Logger;
+
+namespace Sinch.Conversation.TemplatesV2
+{
+    /// <summary>
+    ///     The Template Management API offers a way to manage templates that can be used together with the Conversation API.
+    ///     Note that you may also use the Message Composer tool on the [Sinch Customer
+    ///     Dashboard](https://dashboard.sinch.com/convapi/message-composer) to [manage
+    ///     templates](https://community.sinch.com/t5/Conversation-API/How-do-I-use-Message-Composer-to-create-omni-channel-message/ta-p/9890).
+    ///     One can view a template as a pre-defined message that can optionally contain some parameters to facilitate some
+    ///     customization of the pre-defined message. This feature can, for instance, be used to construct a generic customer
+    ///     welcome message where the customer's name can be injected via a parameter. It's also possible to provide
+    ///     translations to different languages when creating a template to make it possible to reuse one template for
+    ///     different languages.
+    /// </summary>
+    public interface ISinchConversationTemplatesV2
+    {
+        /// <summary>
+        ///     Get a template
+        /// </summary>
+        /// <param name="templateId"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<Template> Get(string templateId, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        ///     List all templates
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<IEnumerable<Template>> List(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        ///     Creates a template
+        /// </summary>
+        /// <param name="template">Template to create</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<Template> Create(CreateTemplateRequest template, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        ///    
+        /// </summary>
+        /// <param name="templateId"></param>
+        /// <param name="languageCode">Optional. The translation's language code.</param>
+        /// <param name="translationVersion">Optional. The translation's version.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<IEnumerable<TemplateTranslation>> ListTranslations(string templateId, string languageCode,
+            string translationVersion,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        ///     Update the template
+        /// </summary>
+        /// <param name="template"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<Template> Update(Template template, CancellationToken cancellationToken = default);
```

Comment:
This object contains a property `id` which is not part of the specification

Reply:
It's required query parameter in path nontheless, but I'll split it to different class